web/src/main/java/org/cbioportal/web/parameter/Session.java

```diff
@@ -0,0 +1,59 @@
+package org.cbioportal.web.parameter;
+
+import java.util.HashMap;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonInclude(Include.NON_NULL)
+public class Session {
+
+    private String id;
+    private String source;
+    private SessionType type;
+    
+    //replace Object with SessionData once model for main_session and comparison_session are finalized
+    private Object data;
```

Comment:
Are we still finalizing the models?

Reply:
I will update for `comparison_session`. could you help in defining for `main_session`?

Reply:
@kalletlak I could help with that. The main session seems quite complicated. But the good news is based on @adamabeshouse  even we do sometimes change the request to create a session but not so frequent, and because we need to make sure the session is backward compatible, so often times, the new change will be adding a new parameter.

We should probably discuss the process that if we need to add a property in the model, this is what I have in mind:
- [ ] Update the model in session service
- [ ] Update the same model in portal backend
- [ ] Update the frontend type and features related

@kalletlak @adamabeshouse @alisman @inodb @jjgao any more comments?

Reply:
Another thing, if @kalletlak you could create the models at session service side and generate the client at the frontend, do we still should add the models at portal backend?

Reply:
After discuss, we decided to keep Session a generic model in session-service. All portal specific models are only added in portal backend.